### ISSUE TITLE
Loguru integration

### ISSUE THREAD

**Issue by bbonenfant** [MORSECorp/snappiershot#37]
Exploratory ticket on the use of loguru within snappiershot.

Some topics to cover:
* Do we want to add a dependency like this (if not could we have this be an optional "feature")?
* How is logging typically done within packages and/or plugins?
* What would we be logging within snappiershot? How would users find and interact with these logs?

**Comment by lenabradley** [comment]
I'm undecided if logging really makes sense here or not (`loguru`-based or otherwise); I go back and forth depending on the day

Random thoughts:
* if we _do_ add logging, it should be off-by-default & something that package-developers can turn on as needed
* things we might want to log: 
    - details about gathering/recording/looking-up snapshot context
    - details about file reading/writing
    - details about which custom en/decoders are being used/called
* while this could all be useful, i'm currently inclined to say we shoudl hold off and add it when we actually start to feel the need for these things.
* `loguru` vs `logging`:
    - `loguru`:
        - pros: easy to use, easy for outside developers to quickly learn & add/modify
        - cons: added dependency, could cause confusion to users
    - `logging`:
        - pros: in standardlib so no added dependency
        - cons: harder to setup initially, maybe harder for some developers to learn & use

**Comment by bbonenfant** [comment]
I agree that any logging should be off-by-default and that we should hold off for now until we see for use cases.
Also, I believe that we should use `logging` over `loguru` to avoid a potentially heavy dependency.
